### PR TITLE
Add cov and sacc scripts + include per-filtering-tag transfer function from clean branch

### PR DESCRIPTION
**Pull request by adrien-laposta**
This PR includes a new feature : the possibility to have different filtering and estimate the associated transfer function. It also includes a preliminary covariance stage as well as the creation of sacc files.

### CODE REVIEW

File: pipeline/pcler.py [hunk 1]
```diff
@@ -118,188 +83,303 @@ def pcler(args):
...
 
-            pcls_mat_filtered = ps_utils.get_pcls_mat_transfer(
-                fields["filtered"],
-                nmt_binning
-            )
-            pcls_mat_unfiltered = ps_utils.get_pcls_mat_transfer(
-                fields["unfiltered"],
-                nmt_binning
-            )
+                    # TO-DO: filter temperature only once !
```
Comment (kwolz) [inline pipeline/pcler.py:278]: Is this still an open issue?

File: pipeline/run_all.sh [hunk 2]
```diff
@@ -40,39 +40,55 @@ python filterer.py --globals ${paramfile} --sims
 echo "Running filterer for data"
 echo "-------------------------"
 python filterer.py --globals ${paramfile} --data
+#OMP_NUM_THREADS=2 mpirun -n 4 python filterer.py --globals ${paramfile} --data
```
Comment (kwolz) [inline pipeline/run_all.sh:43]: My guess is that mpirun is never used at this stage, now that we will use slurm script for toast filtering,

File: soopercool/utils.py [hunk 3]
```diff
@@ -157,6 +123,39 @@ def random_src_mask(mask, nsrcs, mask_radius_arcmin):
     return ps_mask
 
 
+def get_beam_windows(meta, plot=False, beam_floor=1.e-2):
```
Comment (kwolz) [inline soopercool/utils.py:126]: Is beam_floor actually used here?

File: pipeline/transfer_validator.py [hunk 4]
```diff
@@ -33,220 +33,270 @@ def transfer_validator(args):
...
-
-            ax.set_xlim(meta.lmin, meta.lmax)
-
-    plt.savefig(f"{plot_dir}/transfer.pdf", bbox_inches="tight")
+    for ftag1, ftag2 in filtering_pairs:
+        tf_dict = read_transfer(
+            f"{meta.coupling_directory}/transfer_function_{ftag1}x{ftag2}.npz"
+        )
+
+        plt.figure(figsize=(20, 20))
```
Comment (kwolz) [inline pipeline/transfer_validator.py:50]: Should we externalize the TF plotting, e.g. use [this](https://github.com/simonsobs/SOOPERCOOL/blob/887a49a56d38014d89a189c207d6931da3465623/soopercool/utils.py#L427) function?

File: pipeline/transfer_validator.py [hunk 5]
```diff
@@ -33,220 +33,270 @@ def transfer_validator(args):
...
-                    sub.set_xticklabels([])
-                else:
-                    sub.set_xlabel(r"$\ell$", fontsize=13)
-
-                if id2 == 0:
-                    if isinstance(rescaling, float):
-                        main.set_ylabel(r"$C_\ell$", fontsize=13)
+        for ftag1, ftag2 in filtering_pairs:
+            plt.figure(figsize=(16, 16))
+            grid = plt.GridSpec(9, 3, hspace=0.3, wspace=0.3)
```
Comment (kwolz) [inline pipeline/transfer_validator.py:181]: Similar comment to the one above, should we use [this utility](https://github.com/simonsobs/SOOPERCOOL/blob/887a49a56d38014d89a189c207d6931da3465623/soopercool/utils.py#L479) instead?

**Review by kwolz — APPROVED** [review]
Great work! Just a few minor comments.

File: soopercool/utils.py (hunk 3 above)
Comment (adrien-laposta) [reply soopercool/utils.py:126]: No. I will remove it. Beam floor is applied when loading the beams.

File: pipeline/pcler.py (hunk 1 above)
Comment (adrien-laposta) [reply pipeline/pcler.py:278]: We need to include pureT maps to account for filtering induced T-P leakage

File: pipeline/transfer_validator.py (hunk 4 above)
Comment (adrien-laposta) [reply pipeline/transfer_validator.py:50]: Indeed. This function may be move to a separate utils module in the future.

File: pipeline/transfer_validator.py (hunk 5 above)
Comment (adrien-laposta) [reply pipeline/transfer_validator.py:181]: Yes, the plan is to externalize plotting utilities that are used throughout the pipeline (i.e. ps plot with residual)

**Pull request merged**